src/planning/path_planner/include/path_planner/spline.hpp

```diff
@@ -0,0 +1,751 @@
+/*
+ * spline.h
+ *
+ * simple cubic spline interpolation library without external
+ * dependencies https://kluge.in-chemnitz.de/opensource/spline/
+ *
+ * ---------------------------------------------------------------------
+ * Copyright (C) 2011, 2014, 2016, 2021 Tino Kluge (ttk448 at gmail.com)
+ *
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU General Public License
+ *  as published by the Free Software Foundation; either version 2
+ *  of the License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
```

Comment:
@jim-works would you include the [GPL v3 license text](https://www.gnu.org/licenses/gpl-3.0.txt) at [the end of our LICENSE file](https://github.com/Nova-UTD/navigator/blob/360ab836156313d2ffd67ab94012e24a8d8c25ba/LICENSE#L226), along with an explanatory note of which part of our code uses the file, e.g. "Tino Kluge's cubic spline library", including a link to their site (https://kluge.in-chemnitz.de/opensource/spline/)

Reply:
Sure